prover/src/domain.rs

```diff
@@ -51,6 +51,21 @@ impl<B: StarkField> StarkDomain<B> {
         }
     }
 
+    /// Returns a new STARK domain initialized with the provided custom inputs.
+    pub fn from_twiddles(trace_twiddles: Vec<B>, blowup_factor: usize, domain_offset: B) -> Self {
```

Comment:
I know we use this only for tests, but I think it might be a good idea to add a couple of assertions to check that `trace_twiddles.len()` and `blowup_factor` are both powers of 2.